policy-controller/src/main.rs

```diff
@@ -72,13 +77,32 @@ async fn main() -> Result<()> {
             DETECT_TIMEOUT,
         );
 
-        tokio::spawn(index.run(client, ready_tx));
+        tokio::spawn(index.run(client.clone(), ready_tx));
         handle
     };
 
     // Run the gRPC server, serving results by looking up against the index handle.
     tokio::spawn(grpc(grpc_addr, cluster_networks, handle, drain_rx));
 
+    // Run the admission controller
+    if let Some(admission_addr) = admission_addr {
+        let admission = linkerd_policy_controller::admission::Admission(client);
+
+        let routes = warp::path::end()
+            .and(warp::body::json())
+            .and(warp::any().map(move || admission.clone()))
+            .and_then(linkerd_policy_controller::admission::handler)
+            .with(warp::trace::request());
+
+        tokio::spawn(
+            warp::serve(warp::post().and(routes))
+                .tls()
+                .cert_path("/var/run/linkerd/tls/tls.crt")
+                .key_path("/var/run/linkerd/tls/tls.key")
+                .run(admission_addr),
+        );
```

Comment:
we probably need to figure out a nice way to pass the `drain_rx` into the admission-controller server as well (the way we do for the policy gRPC server, etc), so that we can prevent the process from shutting down while we're currently serving an admission webhook request?